### PR TITLE
📁  #330 Allow changing or create the group of a channel (#823)

### PR DESCRIPTION
**Pull request by stephanevieira75**
# Preview 
https://user-images.githubusercontent.com/36481167/106031280-62fa4e00-60cf-11eb-8714-32be6bb8fe8c.mp4

### CODE REVIEW

**Comment by chamerling** [comment]
Having an icon for group (the `+ folder` one) is nice. Anyway, the icon is not the right one, please find some which is more relevant :)

File: twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/ChannelTemplateEditor.tsx [hunk 1]
```diff
@@ -25,17 +28,34 @@ const ChannelTemplateEditor: FC<Props> = ({
...
+      channel_group: group,
       default: defaultChannel,
     });
   });
 
+  const getGroups = (channels: ChannelResource[]) => {
+    const groupsNames: string[] = [];
+    channels
+      .sort((a, b) => (a.data.channel_group || '').localeCompare(b.data.channel_group || ''))
+      .map((channel: ChannelResource) => {
```
Comment (chamerling) [inline twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/ChannelTemplateEditor.tsx:52]: `map` is for transforming the input to output. Here you can use a `forEach` or use `map` like:

```js
const grousNames: string[] = channels.sort(sortFn).map(mapFn).filter(Boolean);
```

**Review by chamerling — CHANGES_REQUESTED** [review]
Almost perfect 🤘🏻

File: twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/ChannelTemplateEditor.tsx (hunk 1 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/ChannelTemplateEditor.tsx]: Updated 👍🏻

**Comment by stephanevieira75** [comment]
Here is the final design, i update the folder icon to a list icon and now show the divider between the icon and the channel name input. 😄 

Capture d’écran de 2021-01-28 11-37-48

**Pull request merged**